### PR TITLE
FALCON-1937 Cluster update documentation

### PR DESCRIPTION
**Pull request by bvellanki**

### CODE REVIEW

File: docs/src/site/twiki/FalconDocumentation.twiki [hunk 1]
```diff
@@ -615,10 +617,35 @@ Failure to follow any of the above rules would result in a process submission fa
 present in the system for the specified time period, the process can be submitted and scheduled, but all instances
 created would result in a WAITING state unless data is actually provided in the cluster.
 
+---++ Updating cluster entity definition
+User would want to update the cluster entity in following scenarios
```
Comment (yzheng-hortonworks) [inline docs/src/site/twiki/FalconDocumentation.twiki]: How about combining the first two sentences: Cluster entities can be updated when the user wants to change their interface endpoints or properties, e.g. hadoop clusters updated from unsecure to secure; hadoop cluster moved from non high-availability to high-availability, etc.

File: docs/src/site/twiki/FalconDocumentation.twiki [hunk 2]
```diff
@@ -615,10 +617,35 @@ Failure to follow any of the above rules would result in a process submission fa
...
+   * when hadoop clusters are updated from unsecure to secure
+   * when hadoop clusters move from non high-availability to high-availability etc
 
+In these scenarios, user would want to change the cluster entity to reflect updated interface endpoints or properties.
+Updating cluster would require cascading update to feed and process entity jobs in the workflow engine. So Falcon restricts
```
Comment (yzheng-hortonworks) [inline docs/src/site/twiki/FalconDocumentation.twiki]: How about: "feed and process entity jobs" -> "dependent feed/process jobs"

File: docs/src/site/twiki/FalconDocumentation.twiki [hunk 3]
```diff
@@ -615,10 +617,35 @@ Failure to follow any of the above rules would result in a process submission fa
...
+Cluster update to following
+   * Cluster update can only be performed by superuser and when Falcon Server is in safemode.
```
Comment (yzheng-hortonworks) [inline docs/src/site/twiki/FalconDocumentation.twiki]: How about: So Falcon only allows cluster update when:
1. Falcon server is in safemode;
2. The update is requested by superuser;
3. ...

File: docs/src/site/twiki/FalconDocumentation.twiki [hunk 4]
```diff
@@ -615,10 +617,35 @@ Failure to follow any of the above rules would result in a process submission fa
...
+   * The underlying namenode or workflow engine referenced by interface URI is the same. It is only the URI that has changed to reflect secure/HA environments.
+
+Cluster entity should be updated by superuser using following CLI command.
+<verbatim>
+bash$ falcon entity -type cluster -name primaryCluster -update -file ~/primary-updated.xml
```
Comment (yzheng-hortonworks) [inline docs/src/site/twiki/FalconDocumentation.twiki]: Feel it is better to refer to CLI tutorial page, so this document provides a high-lever introduction and users can refer to CLI/API tutorials for the command they need. Same for the other CLI command you listed below.

File: docs/src/site/twiki/FalconDocumentation.twiki [hunk 5]
```diff
@@ -249,9 +250,10 @@ entity from the falcon configuration store. Delete operation on an entity would
 no dependent entities on the deleted entity.
 
 ---+++ Update
-Update operation allows an already submitted/scheduled entity to be updated. Cluster update is currently
-not allowed. Feed update can cause cascading update to all the processes already scheduled. Process update triggers
-update in falcon if entity is updated. The following set of actions are performed in scheduler to realize an update:
+Update operation allows an already submitted/scheduled entity to be updated. Cluster update will require user to  update
+ dependent Feed and Process entities that are already scheduled. Feed update can cause cascading update to all the processes
```
Comment (yzheng-hortonworks) [inline docs/src/site/twiki/FalconDocumentation.twiki]: How about: Cluster update needs to be performed in safemode. Users also need to update the scheduled dependent entities after finishing cluster update and exiting safemode.

File: docs/src/site/twiki/FalconDocumentation.twiki [hunk 6]
```diff
@@ -249,9 +250,10 @@ entity from the falcon configuration store. Delete operation on an entity would
...
+already scheduled. Process update triggers update in falcon if entity is scheduled. The following set of actions are
```
Comment (yzheng-hortonworks) [inline docs/src/site/twiki/FalconDocumentation.twiki]: Better to split cluster update and feed/process update to two paragraphs, one for each.

**Comment by bvellanki** [comment]
@yzheng-hortonworks  : I addressed all your comments.

**Comment by yzheng-hortonworks** [comment]
+1

**Pull request closed without merging**